internal/validation/validation.go

```diff
@@ -193,7 +193,8 @@ func validateValue(c *opContext, v *common.InputValue, val interface{}, t common
 		}
 		vv, ok := val.([]interface{})
 		if !ok {
-			c.addErr(v.Loc, "VariablesOfCorrectType", "Variable \"%s\" has invalid type %T.\nExpected type \"%s\", found %v.", v.Name.Name, val, t, val)
+			// Input coercion rules allow single items without wrapping array
```

Comment:
This resolves the issue with input coercion for lists where a single value is supplied in variables, without a wrapping list